src/transformers/modeling_dpr.py

```diff
@@ -0,0 +1,541 @@
+
+        Examples::
+
+            tokenizer = DPRTokenizer.from_pretrained('dpr-base-uncased')
+            model = DPRModel.from_pretrained('dpr-reader-base')
+            question_and_titles_ids = [
+                torch.tensor(tokenizer("Hello, is my dog cute ?", "Dog cuteness"))
+                ]  # One tensor per passage. It corresponds to the concatenation of the question and the context title.
+            texts_ids = [
+                torch.tensor(tokenizer("Hello, my dog is definitely cute", add_special_tokens=False))
+                ]  # One tensor per passage. It corresponds to the context text in which we're looking for the answer.
+            predicted_spans = model.generate(question_and_titles_ids, texts_ids)
+            # get best answer
+            best_span = predicted_spans[0]
+            best_span_ids = texts_ids[best_span.doc_id].numpy().flatten()
+            best_span_ids = best_span_ids[best_span.start_index:best_span.end_index + 1]
+            print(tokenizer.decode(best_span_ids))
+
+        """
+
+        input_ids, start_logits, end_logits, relevance_logits = self.forward(question_and_titles_ids, texts_ids)
+
+        questions_num, docs_per_question = relevance_logits.size()
+        assert (
+            questions_num == 1
+        ), "`.generate` expects a batch size of 1, i.e. with only the input_ids for one question."
+        _, idxs = torch.sort(relevance_logits, dim=1, descending=True,)
+        nbest_spans_predictions: List[DPRReaderOutput] = []
+        for p in range(docs_per_question):
+            doc_id = idxs[0, p].item()
+            sequence_ids = input_ids[0, doc_id]
+            sequence_len = sequence_ids.size(0)
+            # assuming question & title information is at the beginning of the sequence
+            passage_offset = question_and_titles_ids[p].size(1)
+
+            p_start_logits = start_logits[0, doc_id].tolist()[passage_offset:sequence_len]
+            p_end_logits = end_logits[0, doc_id].tolist()[passage_offset:sequence_len]
+            ctx_ids = sequence_ids.tolist()[passage_offset:]
+            best_spans = self._get_best_spans(
+                start_logits=p_start_logits,
+                end_logits=p_end_logits,
+                ctx_ids=ctx_ids,
+                max_answer_length=max_answer_length,
+                doc_id=doc_id,
+                relevance_score=relevance_logits[0, doc_id].item(),
+                top_spans=num_spans_per_passage,
+            )
+            nbest_spans_predictions.extend(best_spans)
+            if len(nbest_spans_predictions) > num_spans:
+                break
+        return nbest_spans_predictions[:num_spans]
+
+    def _get_best_spans(
+        self,
+        start_logits: List,
+        end_logits: List,
+        ctx_ids: List,
+        max_answer_length: int,
+        doc_id: int,
+        relevance_score: float,
+        top_spans: int,
+    ) -> List[DPRReaderOutput]:
```

Comment:
Cf. above, I would move this to the tokenizer. This seems to be all about input/output processing rather than PT/TF-accelerated stuff

Reply:
I'll keep it for the next PR (cf other comment)